cmd/apiProviderUpload_test.go

```diff
@@ -0,0 +1,121 @@
+package cmd
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/apim"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRunApiProviderUpload(t *testing.T) {
+	t.Parallel()
+
+	t.Run("happy path", func(t *testing.T) {
+		file, tmpErr := ioutil.TempFile("", "test.json")
+		if tmpErr != nil {
+			t.FailNow()
+		}
+		defer os.RemoveAll(file.Name()) // clean up
+		filesMock := mock.FilesMock{}
+		filesMock.AddFile(file.Name(), []byte("Test content"))
+		config := getDefaultOptionsForApiProvider()
+		config.FilePath = file.Name()
+		httpClientMock := httpMockAPIM{StatusCode: 201, ResponseBody: ``}
+		apim := apim.APIMBundle{APIServiceKey: config.APIServiceKey, Client: &httpClientMock}
+		// test
+		err := createApiProvider(&config, apim, filesMock.FileRead)
+		// assert
+		if assert.NoError(t, err) {
+			t.Run("check file", func(t *testing.T) {
+				assert.Equal(t, fileExists(file.Name()), true)
```

Comment:
'fileExists' is deprecated. Please use piperutils.Files{} instead

Reply:
fixed